core-rust/state-manager/src/limits.rs

```diff
@@ -162,16 +175,24 @@ impl VertexLimitsTracker {
         self.remaining_substate_write_count -= execution_metrics.substate_write_count;
         self.remaining_substate_write_size -= execution_metrics.substate_write_size;
 
-        if self.remaining_transactions_count == 0
-            || self.remaining_transactions_size == 0
-            || self.remaining_execution_cost_units_consumed == 0
-            || self.remaining_substate_read_count == 0
-            || self.remaining_substate_read_size == 0
-            || self.remaining_substate_write_count == 0
-            || self.remaining_substate_write_size == 0
-        {
-            return Ok(VertexLimitsAdvanceSuccess::VertexFilled);
+        Ok(self.check_filled())
+    }
+}
+
+pub trait HasUnit {
+    fn unit() -> Self;
+}
+
+impl HasUnit for ExecutionMetrics {
+    fn unit() -> Self {
```

Comment:
I feel like `unit` is a bit of a weird concept. It's usually used for the multiplicative identity - but that doesn't make sense in this context.

Really you're after "minimum_for_transaction" - in which case eg `max_wasm_memory_used` is 0 because a transaction doesn't need that.

But in any case, I'm not sure if this is a particularly useful check in its current form? :)